src/WasmDis.ts

```diff
@@ -257,6 +266,55 @@ export class DefaultNameResolver implements INameResolver {
     return '$label' + index;
   }
 }
+
+export class DevToolsNameResolver extends DefaultNameResolver {
+  private readonly _functionNames: string[];
+  private readonly _localNames: string[][];
+  private readonly _memoryNames: string[];
+  private readonly _tableNames: string[];
+  private readonly _globalNames: string[];
+
+  constructor(functionNames: string[], localNames: string[][], memoryNames: string[], tableNames: string[], globalNames: string[]) {
+    super();
+    this._functionNames = functionNames;
+    this._localNames = localNames;
+    this._memoryNames = memoryNames;
+    this._tableNames = tableNames;
+    this._globalNames = globalNames;
+  }
+
+  public getTableName(index: number, isRef: boolean): string {
+    const name = this._tableNames[index];
+    if (!name)
+      return super.getTableName(index, isRef);
+    return isRef ? `$${name}` : `$${name} (;${index};)`;
+  }
+  public getMemoryName(index: number, isRef: boolean): string {
```

Comment:
Minor nit: Put a blank line between your function declarations.

Reply:
Taken care of, thanks Eric!